src/main.rs

```diff
@@ -147,14 +148,38 @@ fn main() {
     let mut chat_input_enabled = false;
 
     'running: for tick in 0.. {
-        // Get player pos
+        // Get Health and Energy
+        let (current_health, max_health) = client.current::<comp::Health>().map_or((0, 0), |health| (health.current(), health.maximum()));
+        let (current_energy, max_energy) = client.current::<comp::Energy>().map_or((0, 0), |energy| (energy.current(), energy.maximum()));
+
+        // Invite Logic
+		let (inviter_uid, invite_kind) = if let Some((inviter_uid, _, _, invite_kind)) = client.invite() {
+    		(Some(inviter_uid), Some(invite_kind))
+		} else {
+    		(None, None)
+		};
+		
+		//Get entity username from UID
+        let inviter_username = if let Some(uid) = inviter_uid {
```

Comment:
```rust
let inviter_username = inviter_uid
    .and_then(|uid| client.state().ecs().read_resource::<UidAllocator>().retrieve_entity_internal(uid.id()))
    .and_then(|entity| client.state().read_storage::<comp::Player>().get(entity))
    .map(|player| player.alias.clone())
    .unwrap_or_default();
```

Rust's `Option`/`Result` combinations are nice and simplify code like this well :)